js/map.js

```diff
@@ -188,23 +219,135 @@ window.map = (function () {
     };
     isPageActive = false;
 
-    var mapPins = document.querySelectorAll('.map__pin:not(.map__pin--main)');
     mapPinsMainElement.style.top = MAP_PIN_TOP + 'px';
     mapPinsMainElement.style.left = MAP_PIN_LEFT + 'px';
     mapElement.classList.add('map--faded');
     adFormContainer.classList.add('ad-form--disabled');
+    filterTypeElement.value = 'any';
+    filterPriceElement.value = 'any';
+    filterRoomsElement.value = 'any';
+    filterGuestsElement.value = 'any';
+    
+    Array.prototype.forEach.call(housingFeaturesCheckboxes, function (elem) {
+      if (elem.checked) {
+        elem.checked = false;
+      }
+    });
+    
+    removeCard();
+    removePins();
+  }
 
-    if (mapElement.querySelector('.popup')) {
-      document.querySelector('.popup__close').removeEventListener('click', closeCardHandler);
-      document.removeEventListener('keydown', escPressCloseCardHandler);
-      removeCard();
-    }
+  function filter() {
+    var type = filterTypeElement.value;
+    var price = filterPriceElement.value;
+    var rooms = filterRoomsElement.value;
+    var guests = filterGuestsElement.value;
+    var featuresArray = [];
 
-    Array.prototype.forEach.call(mapPins, function (elem) {
-      elem.parentElement.removeChild(elem);
+    Array.prototype.forEach.call(housingFeaturesCheckboxes, function (elem) {
+      if (elem.checked) {
+        featuresArray.push(elem.value);
+      }
     });
+
+    var filteredPropertyType;
+    var filteredPropertyPrice;
+    var filteredPropertyRooms;
+    var filteredPropertyGuests;
+    var filteredPropertyFeatures;
+
+    if (filteredData.length > 0) {
+      filteredData.length = 0;
```

Comment:
Why do you need filteredData global? Try to get rid of it. You were planning to make a module out of it. How are you gonna do it with this global var? A function should accept and return values. Avoid globals like that. 
and it's better to `filteredData = []` if you want to reset it.

Reply:
The variable it is used by the ShowCard() function. I made the Filter function to return a result. But in the applyFilter function i save the result to that global variable filteredData as showCard can use it.